pootle/apps/pootle_store/views.py

```diff
@@ -452,7 +453,10 @@ def get_units(request):
     if 'virtualfolder' in settings.INSTALLED_APPS:
         from virtualfolder.helpers import extract_vfolder_from_path
 
-        vfolder, pootle_path = extract_vfolder_from_path(pootle_path)
+        vfolder, pootle_path = extract_vfolder_from_path(
+            pootle_path,
+            vfti=VirtualFolderTreeItem.objects.select_related(
+                "directory", "vfolder"))
```

Comment:
Just wondering if instead of calling here `select_related` it would be better to pass a boolean flag to `extract_vfolder_from_path` and call `select_related` there.